applications/RANSApplication/custom_utilities/rans_variable_utilities.cpp

```diff
@@ -230,6 +234,239 @@ void CopyNodalSolutionStepVariablesList(
 
     KRATOS_CATCH("");
 }
+
+template <typename TDataType>
+void AssignConditionVariableValuesToNodes(ModelPart& rModelPart,
+                                          const Variable<TDataType>& rVariable,
+                                          const Flags& rFlag,
+                                          const bool FlagValue)
+{
+    auto& r_nodes = rModelPart.Nodes();
+    VariableUtils().SetHistoricalVariableToZero(rVariable, r_nodes);
+
+    const int number_of_conditions = rModelPart.NumberOfConditions();
+#pragma omp parallel for
+    for (int i_cond = 0; i_cond < number_of_conditions; ++i_cond)
+    {
+        auto& r_cond = *(rModelPart.ConditionsBegin() + i_cond);
+        if (r_cond.Is(rFlag) == FlagValue)
+        {
+            const int number_of_nodes = r_cond.GetGeometry().PointsNumber();
+            const auto& r_normal = r_cond.GetValue(rVariable);
+            for (int i_node = 0; i_node < number_of_nodes; ++i_node)
+            {
+                auto& r_node = r_cond.GetGeometry()[i_node];
+                r_node.SetLock();
+                r_node.FastGetSolutionStepValue(rVariable) +=
+                    r_normal * (1.0 / static_cast<double>(number_of_nodes));
+                r_node.UnSetLock();
+            }
+        }
+    }
+
+    rModelPart.GetCommunicator().AssembleCurrentData(rVariable);
+}
+
+void AddAnalysisStep(ModelPart& rModelPart, const std::string& rStepName)
+{
+    auto& r_process_info = rModelPart.GetProcessInfo();
+    if (!r_process_info.Has(ANALYSIS_STEPS))
+    {
+        r_process_info.SetValue(ANALYSIS_STEPS, std::vector<std::string>());
+    }
+    r_process_info[ANALYSIS_STEPS].push_back(rStepName);
+}
+
+bool IsAnalysisStepCompleted(const ModelPart& rModelPart, const std::string& rStepName)
+{
+    const auto& r_process_info = rModelPart.GetProcessInfo();
+    if (r_process_info.Has(ANALYSIS_STEPS))
+    {
+        const std::vector<std::string>& r_steps = r_process_info[ANALYSIS_STEPS];
+        return (std::find(r_steps.begin(), r_steps.end(), rStepName) != r_steps.end());
+    }
+    else
+    {
+        return false;
+    }
+}
+
+void AssignBoundaryFlagsToGeometries(ModelPart& rModelPart)
+{
+    const int number_of_conditions = rModelPart.NumberOfConditions();
+#pragma omp parallel for
+    for (int i_cond = 0; i_cond < number_of_conditions; ++i_cond)
+    {
+        Condition& r_condition = *(rModelPart.ConditionsBegin() + i_cond);
+        r_condition.SetValue(RANS_IS_INLET, r_condition.Is(INLET));
+        r_condition.SetValue(RANS_IS_OUTLET, r_condition.Is(OUTLET));
+        r_condition.SetValue(RANS_IS_STRUCTURE, r_condition.Is(STRUCTURE));
```

Comment:
Why do you need specific flags?

Reply:
I create a seperate model part per transport equation on the fly using the same geoemtry and nodes. So all of these model parts' conditions needs to know whether they are INLET, SLIP or STRUCTURE. At the moment we dont have common flag data between conditions which shares geoemtries, but we have common data value container. So I set these value in condition data value container (it is setting these values on corresponding geometry data value container), so all the on-the-fly model part conditions knows what type of condition it is.